File: examples/tests/example-readings-fastapi-pytest-component-test/readings_playbooks.py
from arena_pytest import (
    HttpMappingExpect,
    ManagedHttpPlaybook,
    ManagedLocalstackPlaybook,
    ManagedMssqlPlaybook,
)

CALIBRATION_VALIDATE_PATH = "/api/v1/validate"
LOCALSTACK_SESSION_PLAYBOOK_ID = "readings-api-localstack-session"
CALIBRATION_DEFAULT_PLAYBOOK_ID = "readings-api-calibration-default"
VALIDATION_DB_PLAYBOOK_ID = "readings-api-validation-db-scoped"


class CalibrationDefaultPlaybook(ManagedHttpPlaybook):
    def __init__(self, dependency_identifier: str):
        super().__init__(
            identifier=CALIBRATION_DEFAULT_PLAYBOOK_ID,
            dependency_identifier=dependency_identifier,
            mappings=[
                {
                    "method": "POST",
                    "url_path": CALIBRATION_VALIDATE_PATH,
                    "status": 200,
                    "json_body": {"valid": True},
                    "expect": HttpMappingExpect.called_at_least(1),
                }
            ],
        )


class ValidationDbPlaybook(ManagedMssqlPlaybook):
    def __init__(self, dependency_identifier: str):
        super().__init__(
            identifier=VALIDATION_DB_PLAYBOOK_ID,
            dependency_identifier=dependency_identifier,
        )


class LocalstackSessionPlaybook(ManagedLocalstackPlaybook):
    def __init__(self, dependency_identifier: str):
        super().__init__(
            identifier=LOCALSTACK_SESSION_PLAYBOOK_ID,
            dependency_identifier=dependency_identifier,
        )

File: examples/tests/example-readings-spring-boot-component-test/arena/examples/readings/component/ReadingsPlaybooks.java
package arena.examples.readings.component;

import arena.junit.playbook.ManagedHttpPlaybook;
import arena.junit.playbook.ManagedLocalstackPlaybook;
import arena.junit.playbook.ManagedMssqlPlaybook;
import java.util.List;
import java.util.Map;

final class ReadingsPlaybooks {
  private ReadingsPlaybooks() {}

  static final class CalibrationDefaultPlaybook extends ManagedHttpPlaybook {
    CalibrationDefaultPlaybook(String dependencyIdentifier) {
      super(
          "readings-api-calibration-default",
          dependencyIdentifier,
          List.of(
              mapping(
                  "POST",
                  ReadingsArenaFixture.CALIBRATION_VALIDATE_PATH,
                  200,
                  Map.of("valid", true),
                  Expect.calledAtLeast(1))));
    }
  }

  static final class ValidationDbPlaybook extends ManagedMssqlPlaybook {
    ValidationDbPlaybook(String dependencyIdentifier) {
      super("readings-api-validation-db-scoped", dependencyIdentifier);
    }
  }

  static final class LocalstackSessionPlaybook extends ManagedLocalstackPlaybook {
    LocalstackSessionPlaybook(String dependencyIdentifier) {
      super("readings-api-localstack-session", dependencyIdentifier);
    }
  }
}
